lib/ProductOpener/Products.pm

```diff
@@ -124,6 +124,7 @@ BEGIN {
 
 use vars @EXPORT_OK;
 
+use ProductOpener::ProductSchemaChanges qw/$current_schema_version convert_product_schema/;
```

Comment:
you import convert_product_schema but do not use it

```suggestion
use ProductOpener::ProductSchemaChanges qw/$current_schema_version/;
```

Reply:
I'm thinking to add it in retrieve_product() so that it's always converted to the newest schema